modules/face/src/getlandmarks.cpp

```diff
@@ -0,0 +1,245 @@
+        }
+    }
+    return true;
+}
+void FacemarkKazemiImpl :: readSplit(ifstream& is, splitr &vec)
+{
+    is.read((char*)&vec, sizeof(splitr));
+}
+void FacemarkKazemiImpl :: readLeaf(ifstream& is, vector<Point2f> &leaf)
+{
+    uint64_t size;
+    is.read((char*)&size, sizeof(size));
+    leaf.resize((size_t)size);
+    is.read((char*)&leaf[0], leaf.size() * sizeof(Point2f));
+}
+void FacemarkKazemiImpl :: readPixels(ifstream& is,uint64_t index)
+{
+    is.read((char*)&loaded_pixel_coordinates[(unsigned long)index][0], loaded_pixel_coordinates[(unsigned long)index].size() * sizeof(Point2f));
+}
+void FacemarkKazemiImpl :: loadModel(String filename){
+    if(filename.empty()){
+        String error_message = "No filename found.Aborting....";
+        CV_Error(Error::StsBadArg, error_message);
+        return ;
+    }
+    ifstream f(filename.c_str(),ios::binary);
+    if(!f.is_open()){
+        String error_message = "No file with given name found.Aborting....";
+        CV_Error(Error::StsBadArg, error_message);
+        return ;
+    }
+    uint64_t len;
+    f.read((char*)&len, sizeof(len));
+    char* temp = new char[(size_t)len+1];
+    f.read(temp, len);
+    temp[len] = '\0';
+    string s(temp);
+    delete [] temp;
+    if(s.compare("cascade_depth")!=0){
+        String error_message = "Data not saved properly.Aborting.....";
+        CV_Error(Error::StsBadArg, error_message);
+        return ;
+    }
+    uint64_t cascade_size;
+    f.read((char*)&cascade_size,sizeof(cascade_size));
+    loaded_forests.resize((unsigned long)cascade_size);
+    f.read((char*)&len, sizeof(len));
+    temp = new char[(unsigned long)len+1];
+    f.read(temp, len);
+    temp[len] = '\0';
+    s = string(temp);
+    delete [] temp;
+    if(s.compare("pixel_coordinates")!=0){
+        String error_message = "Data not saved properly.Aborting.....";
+        CV_Error(Error::StsBadArg, error_message);
+        return ;
+    }
+    loaded_pixel_coordinates.resize((unsigned long)cascade_size);
+    uint64_t num_pixels;
+    f.read((char*)&num_pixels,sizeof(num_pixels));
+    for(unsigned long i=0 ; i < cascade_size ; i++){
+        loaded_pixel_coordinates[i].resize((unsigned long)num_pixels);
+        readPixels(f,i);
+    }
+    f.read((char*)&len, sizeof(len));
+    temp = new char[(unsigned long)len+1];
+    f.read(temp, len);
+    temp[len] = '\0';
+    s = string(temp);
+    delete [] temp;
+    if(s.compare("mean_shape")!=0){
+        String error_message = "Data not saved properly.Aborting.....";
+        CV_Error(Error::StsBadArg, error_message);
+        return ;
+    }
+    uint64_t mean_shape_size;
+    f.read((char*)&mean_shape_size,sizeof(mean_shape_size));
+    meanshape.resize((unsigned long)mean_shape_size);
+    f.read((char*)&meanshape[0], meanshape.size() * sizeof(Point2f));
+    if(!setMeanExtreme())
+        exit(0);
+    f.read((char*)&len, sizeof(len));
+    temp = new char[(unsigned long)len+1];
```

Comment:
Consider to use `cv::AutoBuffer<char> temp(len + 1);`

Reply:
General message here is that we should avoid manual memory management via malloc/free/new/delete.

`cv::AutoBuffer` is like `std::vector`.